backend/src/websocket_manager.py

```
@@ -0,0 +1,84 @@
+from fastapi import WebSocket
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ConnectionManager:
+    """Manages WebSocket connections for real-time job updates."""
+    
+    def __init__(self):
+        # Store active connections by job_id
+        self.active_connections: Dict[str, Set[WebSocket]] = {}
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)